### PR TITLE
Add possibility to filter on presence of trigger muons and B candidates

### PR DESCRIPTION
**Pull request by sarafiorendi**
- Added one filter which requires the presence of at least an offline muon matched to a trigger object.
- Moved from one single cms.Path to 2, one for BKee and one for BKmumu, so that is easy to select events with at least N candidates (be it BKmumu or BKee) passing the selection cuts .
By default with this PR, at least 1 candidate is requested.

Running on 500 events, this is what I get:
nCands_afterFilter

### CODE REVIEW

**Comment by mverzett** [comment]
The PR fails in the MC with the error:
```
19-Aug-2019 16:37:06 CEST  Successfully opened file root://eoscms.cern.ch//eos/cms/store/cmst3/group/bpark/BToKmumu_1000Events_MINIAOD.root
----- Begin Fatal Exception 19-Aug-2019 16:37:23 CEST-----------------------
An exception of category 'ScheduleExecutionFailure' occurred while
   [0] Calling beginJob
Exception Message:
Unrunnable schedule
Module run order problem found:
lheInfoTable after rivetLeptonTable [path nanoAOD_KMuMu_step], rivetLeptonTable after genParticleBParkTable [path nanoAOD_KMuMu_step], genParticleBParkTable after genWeightsTable [path nanoAOD_KMuMu_step], genWeightsTable after genTable [path nanoAOD_KMuMu_step], genTable after puTable [path nanoAOD_KMuMu_step], puTable after electronBParkMCTable [path nanoAOD_KMuMu_step], electronBParkMCTable after electronsBParkMCMatchForTable [path nanoAOD_KMuMu_step], electronsBParkMCMatchForTable consumes electronsForAnalysis, electronsForAnalysis after lowPtGsfElectronLatestID [path nanoAOD_Kee_step], lowPtGsfElectronLatestID after lheInfoTable [path nanoAOD_Kee_step]
 Running in the threaded framework would lead to indeterminate results.
 Please change order of modules in mentioned Path(s) to avoid inconsistent module ordering.
----- End Fatal Exception -------------------------------------------------
19-Aug-2019 16:37:23 CEST  Closed file root://eoscms.cern.ch//eos/cms/store/cmst3/group/bpark/BToKmumu_1000Events_MINIAOD.root
```

**Comment by sarafiorendi** [comment]
mmm ok I haven't checked on the MC. Let me take a look.
Also, I think I should also change the cfg for the FEVTDEBUG option.

File: BParkingNano/python/nanoBPark_cff.py [hunk 1]
```diff
@@ -23,7 +23,6 @@
 
 nanoSequence = cms.Sequence(nanoMetadata + 
                             vertexSequence +
```
Comment (amartelli) [inline BParkingNano/python/nanoBPark_cff.py]: I think that the removal of muonBParkSequence from here is causing the crash on MC, 
because nanoAOD_customizeMC is run before the nanoAOD_customizeMuonTriggerBPark
and muons are needed there (in the MC sequence) for muonBParkMC

**Comment by sarafiorendi** [comment]
I modified a bit also the logic of the MC sequence, so that the pure GEN part is run before reconstruction is attempted, and MC matching for electrons is not added to BKMuMu path.

**Comment by mverzett** [comment]
Validation can be found [here](https://mverzett.web.cern.ch/mverzett/bph_plots/FAKE_CI/PR31/validation_data/):
  * The average event size increases, as one can expect, by 17%, but the total number of saved events is reduced by ~50%. Over-all a gain.
  * Timing is roughly unaffected, pointing once again to combinatorics in vertexing, rather than all the previous steps
  * The validation was done with a customized script where I learned that `JaggedArray.content` and `JaggedArray.flatten()` do two slightly different things, and here matters
  * The validation completely fails on MC, because the output of #31 is an empty file, the current head, once masked to mimic this selection is empty too. There might be a problem in our B -> Kµµ sequence

TL;DR: To me is OK, but we have to figure out what's going on with B -> Kµµ, after this PR.
